bundles/org.openhab.automation.jrubyscripting/src/main/java/org/openhab/automation/jrubyscripting/internal/JRubyScriptEngineConfiguration.java

```diff
@@ -124,24 +126,62 @@ void configureScriptEngine(ScriptEngineFactory factory) {
     }
 
     /**
-     * Makes Gem home directory if it does not exist
+     * Gets a single configuration element.
      */
-    private void ensureGemHomeExists() {
-        OptionalConfigurationElement gemHomeConfigElement = CONFIGURATION_PARAMETERS.get(GEM_HOME);
-        if (gemHomeConfigElement == null) {
-            return;
+    private String get(String key) {
+        OptionalConfigurationElement configElement = configurationParameters.get(key);
+
+        return Objects.requireNonNull(configElement).getValue();
+    }
+
+    /**
+     * Gets the concrete gem home to install gems into for this version of JRuby.
+     * 
+     * {RUBY_ENGINE} and {RUBY_VERSION} are replaced with their current actual values.
+     */
+    public String getSpecificGemHome() {
+        String gemHome = get(GEM_HOME_CONFIG_KEY);
+        if (gemHome.isEmpty()) {
+            return gemHome;
         }
-        Optional<String> gemHome = gemHomeConfigElement.getValue();
-        if (gemHome.isPresent()) {
-            File gemHomeDirectory = new File(gemHome.get());
-            if (!gemHomeDirectory.exists()) {
-                logger.debug("gem_home directory does not exist, creating");
-                if (!gemHomeDirectory.mkdirs()) {
-                    logger.warn("Error creating gem_home directory");
-                }
+
+        gemHome = gemHome.replace(RUBY_ENGINE_REPLACEMENT, Constants.ENGINE);
+        gemHome = gemHome.replace(RUBY_ENGINE_VERSION_REPLACEMENT, Constants.VERSION);
+        gemHome = gemHome.replace(RUBY_VERSION_REPLACEMENT, Constants.RUBY_VERSION);
+        return new File(gemHome).toString();
+    }
+
+    /**
+     * Get the base for all possible gem homes.
+     * 
+     * If the configured gem home contains {RUBY_ENGINE} or {RUBY_VERSION},
+     * the path is cut off at that point. This means a single configuration
+     * value will include the gem homes for all parallel-installed ruby
+     * versions.
+     * 
+     */
+    public String getGemHomeBase() {
+        String gemHome = get(GEM_HOME_CONFIG_KEY);
+
+        for (String replacement : REPLACEMENTS) {
+            int loc = gemHome.indexOf(replacement);
+            if (loc != -1) {
+                gemHome = gemHome.substring(0, loc);
+            }
+        }
+        return new File(gemHome).toString();
+    }
+
+    /**
+     * Makes Gem home directory if it does not exist
+     */
+    private void ensureGemHomeExists(String gemHome) {
+        File gemHomeDirectory = new File(gemHome);
+        if (!gemHomeDirectory.exists()) {
+            logger.debug("gem_home directory does not exist, creating");
+            if (!gemHomeDirectory.mkdirs()) {
+                logger.warn("Error creating gem_home directory");
```

Comment:
Should the throw an exception to be handled or return false, so caller would be notified and could avoid continuing when this directory doesn't exist, possibly running into problems?

Reply:
+1, I can add.